lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -717,10 +717,15 @@ prop_randomOpChunks (KeyValPairs pairs) =
             unsafeRunExceptT $ putCheckpoint k cp
             unsafeRunExceptT $ putWalletMeta k meta
         else do
-            atomically $ unsafeRunExceptT $ initializeWallet k cp meta mempty gp
+            let cp0 = imposeGenesisState cp
+            atomically $ unsafeRunExceptT $ initializeWallet k cp0 meta mempty gp
             Set.fromList <$> atomically listWallets
                 `shouldReturn` Set.fromList (k:keys)
 
+    imposeGenesisState :: Wallet s -> Wallet s
```

Comment:
or maybe `enforce` ? ;-) `impose` is also good and conveys idea here

Reply:
May the `impose` be with you! :D